### PR TITLE
Documented the Java and JS variable name setting

### PR DESCRIPTION
**Pull request by EnasAbdelrazek**
It was implemented in 9.23 but we missed adding it to the docs.

### CODE REVIEW

File: content/en/docs/refguide/modeling/resources/java-actions.md [hunk 1]
```diff
@@ -65,12 +65,20 @@ For each parameter in the template, define a microflow expression, the value of
...
+
+#### 2.3.1 Return Type
 
 The return type determines the type of the data that the Java action returns. It corresponds with the return type of the `executeAction()` method in the *.java* file of the Java action. You can use the result of a Java action in the microflow in which you call it. See [Data Types](/refguide/data-types/) for the possible return types.
 
 As with parameters, the return type can also be an object or a list of some type parameter. The type parameter you choose for the return type must also be used by at least one of the Java action parameters.
 
+#### 2.3.2 Variable Name
+
+This setting allows you to give a name to the Java action's return value if a return type is selected. This name is used when you drag the action into a microflow. The default value is set to **ReturnValueName**.
```
Comment (MarkvanMents) [inline content/en/docs/refguide/modeling/resources/java-actions.md:80]: I've changed the wording here as this only works if you drag your action into the flow. If you use the **Toolbox** to add your action, then the variable name is ignored.
Is this a bug? I used 10.10 to test it.

File: content/en/docs/refguide9/modeling/resources/java-actions.md [hunk 2]
```diff
@@ -65,12 +65,24 @@ For each parameter in the template, define a microflow expression, the value of
...
 The return type determines the type of the data that the Java action returns. It corresponds with the return type of the `executeAction()` method in the *.java* file of the Java action. You can use the result of a Java action in the microflow in which you call it. See [Data Types](/refguide9/data-types/) for the possible return types.
 
 As with parameters, the return type can also be an object or a list of some type parameter. The type parameter you choose for the return type must also be used by at least one of the Java action parameters.
 
+#### 2.3.2 Variable Name
+
+This setting allows you to give a name to the Java action's return value if a return type is selected. This name is used when you drag the action into a microflow. The default value is set to **ReturnValueName**.
+
+{{% alert color="info" %}}
+The ability to name return variables was added in Mendix version 9.23.0
```
Comment (MarkvanMents) [inline content/en/docs/refguide9/modeling/resources/java-actions.md:83]: Added the version when this feature was added. Taken from Release Notes.

**Review by MarkvanMents — CHANGES_REQUESTED** [review]
@EnasAbdelrazek 
Thanks for the PR, I've reviewed it and have a couple of comments below.
Is it a bug that the variable name is ignored if you use the toolbox to add the action?
Also, I've removed the images as they didn't add much to the documentation - they work in a standard way.
Please can you review and let me know if my changes are OK?
Thanks.

File: content/en/docs/refguide/modeling/resources/java-actions.md (hunk 1 above)
Comment (EnasAbdelrazek) [reply content/en/docs/refguide/modeling/resources/java-actions.md:80]: Hi Mark, thanks for your finding. I tested it as well and it appears we missed it. We can improve this later.

**Review by MarkvanMents — APPROVED** [review]
@EnasAbdelrazek 
Thanks for your feedback.
I'm approving now - is it OK to merge immediately?

**Review by exec-astraea — APPROVED** [review]
Hey Enas! I see no issues from the technical side 👍

**Comment by EnasAbdelrazek** [comment]
Thanks all for reviewing it. Yes, @MarkvanMents You can merge it.

**Pull request merged**